models/Device.js

```diff
@@ -279,6 +279,7 @@ module.exports = function(sequelize, DataTypes) {
    */
   Device.queryDevices = async function(authUser, devices, groups, operator) {
```

Comment:
It would be clearly if the various parts of this function were extracted into smaller methods. It's pretty long and hairy.

For this PR or later, I don't mind.